src/app/shared/services/permission-proxy.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { PermissionsRpcTopic } from '@onecx/integration-interface';
+import { PermissionsCacheService } from '@onecx/shell-core';
+import { filter, firstValueFrom, map } from 'rxjs';
+import { PermissionBffService } from '../generated';
+
+@Injectable({ providedIn: 'root' })
+export class PermissionProxyService {
+  private permissionsTopic$ = new PermissionsRpcTopic();
+
+  constructor(
+    private permissionsService: PermissionBffService,
+    private permissionsCacheService: PermissionsCacheService
+  ) {}
+
+  async init(): Promise<unknown> {
+    this.permissionsTopic$
+      .pipe(
+        filter(
+          (message) =>
+            message.permissions === undefined || message.permissions.length == 0
```

Comment:
Remove message.permissions.length == 0 otherwise you will have an endless loop when the user has no permissions